packages/navigator/src/browser/throttle.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from 'inversify';
+import { Event, Emitter } from '@theia/core/lib/common/event';
+import { Disposable, DisposableCollection } from '@theia/core/lib/common/disposable';
+
+/**
+ * Options for the search term throttle.
+ */
+@injectable()
+export class ThrottleOptions {
+
+    /**
+     * The delay (in milliseconds) before the throttle notifies clients about its content change.
+     */
+    readonly delay: number;
+
+}
+
+export namespace ThrottleOptions {
+
+    /**
+     * The default throttle option.
+     */
+    export const DEFAULT: ThrottleOptions = {
+        delay: 50
+    };
+
+}
+
+/**
+ * The search term throttle. It notifies clients if the underlying search term has changed after a given
+ * amount of delay.
+ */
+@injectable()
+export class Throttle implements Disposable {
```

Comment:
It sounds very generic. Should we move it to the core? Could it be replaced by `lodash.throttle`? Or we rename it to something more specific for the navigator?

Reply:
I will rename it and leave it here.